engines/python/setup/djl_python/request_output.py

```diff
@@ -0,0 +1,132 @@
+
+
+@dataclass
+class Sequence:
+    """Dataclass to store the generated tokens details of all the sequences in the request.
+    Attributes:
+        tokens: generated tokens of the sequence.
+        top_tokens: top tokens of the sequence.
+        cumulative_log_prob: cumulative log probability of the sequence.
+        finish_reason: finish reason of the sequence.
+        stop_reason: stop reason of the sequence.
+        generated_text: generated text of the sequence.
+    """
+    tokens: List[Token] = field(default_factory=lambda: [])
+    top_tokens: Optional[List[List[Token]]] = field(default_factory=lambda: [])
+    cumulative_log_prob: float = 0.0
+    finish_reason: str = False
+    stop_reason: Optional[str] = None
+    generated_text: str = ""
+
+    def set_next_token(self, token: Token):
+        self.tokens.append(token)
+        self.generated_text += token.text
+
+    def set_next_top_tokens(self, top_tokens: List[Token]):
+        self.top_tokens.append(top_tokens)
+
+    @property
+    def next_token(self):
+        return self.tokens[-1]
+
+    @property
+    def next_top_tokens(self):
+        return self.top_tokens[-1] if self.top_tokens else None
+
+    def tokens_as_dict(self):
+        return [token.as_dict() for token in self.tokens]
+
+    def top_tokens_as_dict(self):
+        top_tokens = []
+        for top_token in self.top_tokens:
+            top_tokens.append([token.as_dict() for token in top_token])
+        return top_tokens
+
+
+@dataclass
+class RequestInput:
+    """Base class for all request inputs.
+    Attributes:
+        request_id: The request ID.
+        output_formatter: Output formatter of the request
+        parameters: parameters in the request payload
+    """
+    request_id: int
+    output_formatter: Union[Callable, str] = None
+    parameters: Dict = field(default_factory=lambda: [])
+
+
+@dataclass
+class TextInput(RequestInput):
+    """Request input for text generation.
+    Attributes:
+        input_text: The input text.
+        input_ids: The input tokens ids.
+        adapter: adapter of the request.
+        tokenizer: tokenizer of the model.
+    """
+    input_text: str = None
+    input_ids: List[int] = field(default_factory=lambda: [])
+    adapter: Optional[Any] = None
+    tokenizer: Optional[Any] = None
+
+    @property
+    def prompt_tokens_length(self) -> int:
+        return len(self.input_ids)
+
+
+@dataclass
+class RequestOutput:
+    """Base class for all request outputs.
+    Attributes:
+        request_id: The request ID.
+        input: The request input.
+        prompt_tokens_details: prompt tokens details such as their log_probs and token_ids.
+        finished: Whether the request is finished.
+    """
+    request_id: int
+    input: RequestInput
+    prompt_tokens_details: List[dict] = field(default_factory=lambda: {})
+    finished: bool = False
+
+
+@dataclass
+class TextGenerationOutput(RequestOutput):
```

Comment:
when will we use this class?

Reply:
This class should be used to store all the text generation related output, which is the only one we have right now.